packages/app/src/background/services/wallet/index.ts

```diff
@@ -72,34 +73,41 @@ export default class WalletService implements IBackupable {
       throw new Error("Generate mnemonic first");
     }
 
-    const mnemonic = this.cryptoService.decrypt(encryptedMnemonic);
+    const mnemonic = this.cryptoService.decrypt(encryptedMnemonic, { mode: ECryptMode.PASSWORD });
     const wallet = Wallet.fromPhrase(mnemonic);
     const rawData = await this.accountStorage.get<string>();
+    this.cryptoService.setMnemonic(mnemonic);
```

Comment:
I dont see the point of using `this.cryptoService.setMnemonic(mnemonic);`
Is the mnemonic changed in order to reset it again in the CryptoService in the `generateKeyPair` function. 
Because `generateMnemonic` has already this step to set mnemonic inside the CryptoService.

Reply:
Generate mnemonic and generate key pair are two separate function and can be called when user session is expired so this is why mnemonic is set inside two functions.